src/Avalonia.Visuals/Media/BoxShadow.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Globalization;
+using Avalonia.Animation.Animators;
+using Avalonia.Utilities;
+
+namespace Avalonia.Media
+{
+    public struct BoxShadow
+    {
+        public double OffsetX { get; set; }
+        public double OffsetY { get; set; }
+        public double Blur { get; set; }
+        public double Spread { get; set; }
+        public Color Color { get; set; }
+        public bool IsInset { get; set; }
+
+        static BoxShadow()
+        {
+            Animation.Animation.RegisterAnimator<BoxShadowAnimator>(prop =>
+                typeof(BoxShadow).IsAssignableFrom(prop.PropertyType));
+        }
+        
+        public bool Equals(BoxShadow other)
+        {
+            return OffsetX.Equals(other.OffsetX) && OffsetY.Equals(other.OffsetY) && Blur.Equals(other.Blur) && Spread.Equals(other.Spread) && Color.Equals(other.Color);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is BoxShadow other && Equals(other);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = OffsetX.GetHashCode();
+                hashCode = (hashCode * 397) ^ OffsetY.GetHashCode();
+                hashCode = (hashCode * 397) ^ Blur.GetHashCode();
+                hashCode = (hashCode * 397) ^ Spread.GetHashCode();
+                hashCode = (hashCode * 397) ^ Color.GetHashCode();
+                return hashCode;
+            }
+        }
+
+        public bool IsEmpty => OffsetX == 0 && OffsetY == 0 && Blur == 0 && Spread == 0;
+
+        private readonly static char[] s_Separator = new char[] { ' ', '\t' };
+
+        struct ArrayReader
+        {
+            private int _index;
+            private string[] _arr;
+
+            public ArrayReader(string[] arr)
+            {
+                _arr = arr;
+                _index = 0;
+            }
+
+            public bool TryReadString(out string s)
+            {
+                s = null;
+                if (_index >= _arr.Length)
+                    return false;
+                s = _arr[_index];
+                _index++;
+                return true;
+            }
+
+            public string ReadString()
+            {
+                if(!TryReadString(out var rv))
+                    throw new FormatException();
+                return rv;
+            }
+        }
+        public static unsafe BoxShadow Parse(string s)
+        {
+            if(s == null)
+                throw new ArgumentNullException();
+            if (s.Length == 0)
+                throw new FormatException();
+            if (s[0] == ' ' || s[s.Length - 1] == ' ')
+                s = s.Trim();
```

Comment:
If you use StringSplitOptions.RemoveEmptyEntries, you don't need to make additional Trim before. Also it will handle '\t' on start and end of string (according s_Separator array values).